### PR TITLE
Add connection pool related config options to TestKit back end

### PR DESCRIPTION
**Pull request by robsdedude**
* `connectionAcquisitionTimeoutMs`
 * `maxConnectionPoolSize`

along with the according feature flags

### CODE REVIEW

**Comment by robsdedude** [comment]
When ok, the PR should probably also be back ported to 4.4.

File: testkit-backend/backend.go [hunk 1]
```diff
@@ -392,6 +392,13 @@ func (b *backend) handleRequest(req map[string]interface{}) {
 			if data["resolverRegistered"].(bool) {
 				c.AddressResolver = b.customAddressResolverFunction()
 			}
+			if data["connectionAcquisitionTimeoutMs"] != nil {
+				timeout := time.Millisecond * time.Duration(data["connectionAcquisitionTimeoutMs"].(float64))
```
Comment (fbiville) [inline testkit-backend/backend.go]: nit: you can probably inline the variable for consistency's sake

**Review by fbiville — APPROVED** [review]
🙏

File: testkit-backend/backend.go (hunk 1 above)
Comment (robsdedude) [reply testkit-backend/backend.go]: I could. The line just got ugly long. Do you prefer it in one line?

Comment (fbiville) [reply testkit-backend/backend.go]: Go for it! 
long line

Comment (robsdedude) [reply testkit-backend/backend.go]: Done

it's long now!

**Pull request merged**